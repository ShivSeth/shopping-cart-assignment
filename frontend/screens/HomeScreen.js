import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CategoryDetails from '../components/CategoryDetails';
import { fetchCategory } from '../actions/categoryActions';
import { fetchBanners } from '../actions/bannersActions';
import Carousel from '../components/Carousel';
import { isEmpty } from 'lodash';
const HomeScreen = () => {
  const dispatch = useDispatch();
  const category = useSelector((state) => state.category);
  const banners = useSelector((state) => state.banners);
  let { error, loading, categoryList } = category;

  useEffect(() => {
    dispatch(fetchCategory());
    dispatch(fetchBanners());
  }, [dispatch]);

  return categoryList ? (
    <>
      {!isEmpty(banners) && <Carousel banners={banners} />}
      <CategoryDetails categoryList={categoryList} />
    </>
  ) : error ? (
    <p>Error Occurred {error.toString()}</p>
  ) : (
    <p>Loading</p>
  );
};
export default HomeScreen;
