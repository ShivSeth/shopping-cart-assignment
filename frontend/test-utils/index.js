import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { AuthContextProvider } from '../context/AuthContext';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import store from '../store';
store;
const AllTheProviders = ({ children }) => {
  const history = createMemoryHistory();

  return (
    <Provider store={store}>
      <AuthContextProvider>
        <Router history={history}>{children}</Router>
      </AuthContextProvider>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
