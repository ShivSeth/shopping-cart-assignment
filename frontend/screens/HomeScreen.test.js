import { render, screen } from '../test-utils';
import HomeScreen from './HomeScreen';
describe('<HomeScreen/>', () => {
  test('Explore buttons should be present', async () => {
    const { container } = render(<HomeScreen />);
    const ExploreButton = await screen.findByRole('button', {
      name: 'Explore Beverages',
    });
    const BakeryButton = await screen.findByRole('button', {
      name: 'Explore Bakery Cakes and Dairy',
    });
    const BeautyButton = await screen.findByRole('button', {
      name: 'Explore Beauty and Hygiene',
    });

    expect(ExploreButton).toBeInTheDocument();
    expect(BeautyButton).toBeInTheDocument();
    expect(BakeryButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
