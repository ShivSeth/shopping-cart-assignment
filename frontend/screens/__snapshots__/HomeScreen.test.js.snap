// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> Explore buttons should be present 1`] = `
<div>
  <div
    class="carousel"
    tabindex="0"
  >
    <button
      class="carousel_prev--button"
      type="button"
    >
      PREV
    </button>
    <img
      alt="Independence Day Deal - 25% off on shampoo"
      src="/static/images/offers/offer1.jpg"
    />
    <button
      class="carousel_next--button"
      type="button"
    >
      NEXT
    </button>
    <div
      class="carousel-dots"
    >
      <div
        class="carousel-dot active"
      />
      <div
        class="carousel-dot"
      />
      <div
        class="carousel-dot"
      />
    </div>
    <div
      class="bottom-gradient"
    />
  </div>
  <div
    class="category-details"
  >
    <div
      class="category"
    >
      <img
        alt="Beverages"
        src="/static/images/category/beverages.png"
      />
      <div
        class="category-info"
      >
        <h2>
          Beverages
        </h2>
        <p>
          Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. 
        </p>
        <button>
          Explore Beverages
        </button>
      </div>
      <div
        class="bottom-gradient"
      />
    </div>
    <div
      class="category"
    >
      <img
        alt="Bakery Cakes and Dairy"
        src="/static/images/category/bakery.png"
      />
      <div
        class="category-info"
      >
        <h2>
          Bakery Cakes and Dairy
        </h2>
        <p>
          The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.
        </p>
        <button>
          Explore Bakery Cakes and Dairy
        </button>
      </div>
      <div
        class="bottom-gradient"
      />
    </div>
    <div
      class="category"
    >
      <img
        alt="Beauty and Hygiene"
        src="/static/images/category/beauty.png"
      />
      <div
        class="category-info"
      >
        <h2>
          Beauty and Hygiene
        </h2>
        <p>
          Buy beauty and personal care products online in India at best prices.
        </p>
        <button>
          Explore Beauty and Hygiene
        </button>
      </div>
      <div
        class="bottom-gradient"
      />
    </div>
  </div>
</div>
`;
